### PR TITLE
Anchor idl methods

### PR DESCRIPTION
**Pull request by gchatz22**

### CODE REVIEW

File: src/programs/receiptManager/transaction.ts [hunk 1]
```diff
@@ -205,48 +235,55 @@ export const withClaimRewardReceipt = async (
...
       rewardReceipt: rewardReceiptId,
+      receiptManager: receiptManagerId,
       stakeEntry: params.stakeEntryId,
       receiptEntry: receiptEntryId,
       paymentManager: checkReceiptManager.parsed.paymentManager,
       feeCollectorTokenAccount: feeCollectorTokenAccountId,
       paymentRecipientTokenAccount: paymentRecipientTokenAccountId,
       payerTokenAccount: payerTokenAccountId,
-      payer: params.payer,
+      payer: wallet.publicKey,
```
Comment (jpbogle) [inline src/programs/receiptManager/transaction.ts:255]: should be `params.payer ?? wallet.publicKey` i think

File: src/programs/receiptManager/transaction.ts [hunk 2]
```diff
@@ -126,25 +138,43 @@ export const withUpdateReceiptManager = (
...
-): [Transaction, PublicKey] => {
+): Promise<[Transaction, PublicKey]> => {
   const receiptManagerId = findReceiptManagerId(
     params.stakePoolId,
     params.name
   );
+  const receiptManagerData = await tryGetAccount(() =>
+    getReceiptManager(connection, receiptManagerId)
+  );
+  if (!receiptManagerData) throw "No receipt manager found";
```
Comment (jpbogle) [inline src/programs/receiptManager/transaction.ts:150]: im cool with this i think just not sure if u had a strong reason for doing the account lookup first

**Review by jpbogle — APPROVED** [review]
2 questions otherwise lgtm

File: src/programs/receiptManager/transaction.ts (hunk 2 above)
Comment (gchatz22) [reply src/programs/receiptManager/transaction.ts:150]: yah youre right, i feel like i was a bit hasty w this one. the general approach we have been taking is that if you wanna update some account you have to provide all fields either new or existing ones so we should stick with it. changing …

File: src/programs/receiptManager/transaction.ts (hunk 1 above)
Comment (gchatz22) [reply src/programs/receiptManager/transaction.ts:255]: done

**Pull request merged**